src/main/scala/mesosphere/marathon/core/deployment/impl/DeploymentManagerActor.scala

```diff
@@ -342,6 +344,7 @@ class DeploymentManagerActor(
       plan.id
     )
     runningDeployments.update(plan.id, runningDeployments(plan.id).copy(ref = Some(ref), status = DeploymentStatus.Deploying))
+    deploymentStatus.update(plan.id, DeploymentStepInfo(plan, DeploymentStep.initial, 0))
```

Comment:
Ok, I like everything about this PR but this here is suspicious: the step with index `0` will be overridden when the actual deployment starts?

Reply:
yes. that happens here `case stepInfo: DeploymentStepInfo => deploymentStatus += stepInfo.plan.id -> stepInfo`

Reply:
Do we surprise users by overriding immutable steps? Is it, may be worth taking the first step of the actual plan and put into the `deploymentStatus` instead of `DeploymentStep.initial`?

Reply:
Since we discussed this on slack, I am posting this for other reviewers:

See how's that stuff serialized - https://github.com/mesosphere/marathon/blob/master/src/main/scala/mesosphere/marathon/api/v2/json/Formats.scala#L195 - so basically the API will have stepIndex = 0 and empty actions list...